### PR TITLE
Fix some issues in PlaneWaveTools Utility class

### PR DESCRIPTION
**Pull request by isbeorn**

### CODE REVIEW

File: Utility/Utilities.cs [hunk 1]
```diff
@@ -40,30 +42,35 @@ public partial class Utilities {
...
-                while (i < 4) {
-                    await Task.Delay(TimeSpan.FromSeconds(1), ct);
-                    Logger.Error($"HTTP request to {request.RequestUri} failed: {ex.Message}. Retry attempt {i}");
-
+            while (i < 4) {
+                try {
                     response = await client.SendAsync(request, ct);
+                    break;
+                } catch (WebException ex) {
+                    Logger.Error($"HTTP request to {request.RequestUri} failed: {ex.Message}. Retry attempt {i}");
```
Comment (isbeorn) [inline Utility/Utilities.cs:63]: The retry logic wasn't working when the second attempt would throw an error. rearranging it should fix it.

File: Utility/Utilities.cs [hunk 2]
```diff
@@ -87,18 +94,20 @@ public partial class Utilities {
             var response = await HttpRequestAsync(host, port, "/status", HttpMethod.Get, string.Empty, ct);
             var status = await response.Content.ReadAsStringAsync(ct);
 
-            var dic = StatusRegex().Matches(status).Cast<Match>().ToDictionary(x => x.Groups[1].Value.Trim(), x => x.Groups[2].Value.Trim());
+            var dic = StatusRegex()
+                .Matches(status)
```
Comment (isbeorn) [inline Utility/Utilities.cs:98]: formatted for readability

File: Utility/Utilities.cs [hunk 3]
```diff
@@ -109,18 +118,16 @@ public partial class Utilities {
 
         public static async Task<bool> TestTcpPort(string host, ushort port) {
             bool success = false;
-            Socket s = null;
 
+            using var s = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```
Comment (isbeorn) [inline Utility/Utilities.cs:122]: when having disposable resources the using statement can just be used

File: Utility/Utilities.cs [hunk 4]
```diff
@@ -87,18 +94,20 @@ public partial class Utilities {
...
         }
 
-        public static bool Pwi4CheckMountConnected(string host, ushort port, CancellationToken ct) {
-            var status = new Dictionary<string, string>();
-
-            Task.Run(async () => {
-                status = await Pwi4GetStatus(host, port, ct);
-            }, ct).Wait(ct);
+        public static async Task<bool> Pwi4CheckMountConnected(string host, ushort port, CancellationToken ct) {
+            var status = await Task.Run(() => Pwi4GetStatus(host, port, ct));
```
Comment (isbeorn) [inline Utility/Utilities.cs:110]: .Wait was unnecessary. just await the task and mark the method as async.

Comment (isbeorn) [reply Utility/Utilities.cs:110]: If you don't need it to run a separate task here you can also remove the Task.Run. I just kept this as i'm not sure in which context the caller thread runs

**Comment by daleghent** [comment]
Thanks for catching these.

**Pull request merged**